module/os/freebsd/zfs/zvol_os.c

```diff
@@ -1477,7 +1477,12 @@ zvol_os_create_minor(const char *name)
 		zil_close(zv->zv_zilog);
 	zv->zv_zilog = NULL;
 
-	/* TODO: prefetch for geom tasting */
+	len = MIN(zvol_prefetch_bytes, SPA_MAXBLOCKSIZE);
+	if (len > 0) {
+		dmu_prefetch(os, ZVOL_OBJ, 0, 0, len, ZIO_PRIORITY_ASYNC_READ);
+		dmu_prefetch(os, ZVOL_OBJ, 0, volsize - len, len,
+		    ZIO_PRIORITY_ASYNC_READ);
```

Comment:
I don't think we need this prefetch in case of ZFS_VOLMODE_NONE, since nothing will be able to access it.  Linux seems to not reach this code in similar case.

Reply:
I've decided to open another PR for these purposes https://github.com/openzfs/zfs/pull/17405